### PR TITLE
fix: Conflict with new version of `connectivity_plus` dependency

### PR DESCRIPTION
**Pull request by mbfakourii**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

## Issue

Closes: #986

## Approach
Upgrade flutter dependencies

## Tasks

- [ ] Add tests
- [x] Add changes to documentation (guides, repository pages, code comments)

### CODE REVIEW

**Review by mtrezza — COMMENTED** [review]
Looks good, is this really just a refactor (no release will be published with this PR)?

**Comment by mbfakourii** [comment]
> Looks good, is this really just a refactor (no release will be published with this PR)?

To solve issues number #986, I think it is better to release a new version.

**Comment by mtrezza** [comment]
To release a new version, could you please as usual increase the version in the package and change the PR title to `fix` if it's a bug fix?

**Comment by mbfakourii** [comment]
> To release a new version, could you please as usual increase the version in the package and change the PR title to `fix` if it's a bug fix?

Done

**Comment by mtrezza** [comment]
Could you please review the PR title - if it's a `fix` it should describe the bug; if it's not a bug fix it may be a `feat`, for example to introduce compatibility.

**Comment by mbfakourii** [comment]
> Could you please review the PR title - if it's a `fix` it should describe the bug; if it's not a bug fix it may be a `feat`, for example to introduce compatibility.

Changed.

File: packages/flutter/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## [7.0.1](https://github.com/parse-community/Parse-SDK-Flutter/compare/flutter-7.0.0...flutter-7.0.1) (2024-02-01)
+
+### Bug Fixes
+
+* Upgrade flutter dependencies ([#987](https://github.com/parse-community/Parse-SDK-Flutter/pull/987))
```
Comment (mtrezza) [inline packages/flutter/CHANGELOG.md:5]:
```suggestion
## [7.0.1](https://github.com/parse-community/Parse-SDK-Flutter/compare/flutter-7.0.0...flutter-7.0.1) (2024-02-03)

### Bug Fixes

* Conflict with new version of `connectivity_plus` dependency ([#987](https://github.com/parse-community/Parse-SDK-Flutter/pull/987))
```

File: packages/flutter/pubspec.yaml [hunk 2]
```diff
@@ -25,31 +25,31 @@ dependencies:
   flutter:
     sdk: flutter
 
-  parse_server_sdk: ^5.1.3
+  parse_server_sdk: ^6.3.0
```
Comment (mtrezza) [inline packages/flutter/pubspec.yaml:28]: This looks odd; it's incrementing by a major version; is this a breaking change?

Comment (mbfakourii) [reply packages/flutter/pubspec.yaml:28]: No, it's just updated and not a breaking change

Comment (mbfakourii) [reply packages/flutter/pubspec.yaml:28]: The reason for using this dependency is to use `parse_server_sdk` in `parse_server_sdk_flutter` and the reason for upgrading is to use the latest version of `parse_server_sdk`.

Comment (mtrezza) [reply packages/flutter/pubspec.yaml:28]: OK, so upgrading parse_server_sdk to a higher major version is only then no breaking change for this SDK, if the upgrade doesn't cause any changes for the developer in this SDK's APIs; and if it isn't a transient dependency a developer can use, which could also require code changes for the developer.

Comment (mbfakourii) [reply packages/flutter/pubspec.yaml:28]: According to [this](https://github.com/parse-community/Parse-SDK-Flutter/releases/tag/dart-6.0.0) and review, I think breaking change will happen, maybe it is better not to upgrade this section, what do you think?

Comment (mtrezza) [reply packages/flutter/pubspec.yaml:28]: Not sure, what do you suggest would be best?

Comment (mbfakourii) [reply packages/flutter/pubspec.yaml:28]: We have two ways

1. Bump the Flutter version (breaking change)
2. Return this part to its previous state

My suggestion is to return to the previous state and upgrade with a big feature in the future

Comment (mtrezza) [reply packages/flutter/pubspec.yaml:28]: Sure, we'll go with your decision.

Comment (mbfakourii) [reply packages/flutter/pubspec.yaml]: Reverted to previous version.

**Review by mtrezza — APPROVED** [review]
Looks good!

**Pull request merged**